srcts/src/shiny/initedMethods.ts

```diff
@@ -5,10 +5,33 @@ import type { EventPriority } from "../inputPolicies";
 import type { BindScope } from "./bind";
 import type { Handler, ShinyApp } from "./shinyapp";
 
-let fullShinyObj: Shiny = null;
+let fullShinyObj: FullShinyDef;
+
+// TODO-future; It would be nice to have a way to export this type value instead of / in addition to `Shiny`
+type FullShinyDef = Required<
+  Pick<
+    Shiny,
+    | "bindAll"
+    | "forgetLastInputValue"
+    | "initializeInputs"
+    | "oncustommessage"
+    | "setInputValue"
+    | "shinyapp"
+    | "unbindAll"
+    | "user"
+  >
+> &
+  Shiny;
 
 function setShinyObj(shiny: Shiny): void {
-  fullShinyObj = shiny;
+  fullShinyObj = shiny as FullShinyDef;
+}
+
+function validateShinyHasBeenSet(): FullShinyDef {
+  if (typeof fullShinyObj === "undefined") {
+    throw "Shiny has not finish initialization yet. Please wait for the 'shiny-initialized' event.";
+  }
```

Comment:
Throw an error if internal methods try to utilize methods that have not been initialized yet